WordPress/Classes/ViewRelated/Tools/HelpshiftPresenter.swift

```diff
@@ -0,0 +1,95 @@
+
+
+    /// Presents a Helpshift window displaying a specific FAQ.
+    ///
+    /// - Parameters:
+    ///   - faqID: The 'publish ID' of the FAQ to display.
+    ///   - viewController: The view controller from which to present the Helpshift window.
+    ///   - completion: Optional block to be called when the window is presented.
+    func presentHelpshiftWindowForFAQ(faqID: String, fromViewController viewController: UIViewController, completion: (() -> Void)?) {
+        prepareToDisplayHelpshiftWindow(false) {
+            HelpshiftSupport.showSingleFAQ(faqID, withController: viewController, withOptions: self.optionsDictionary)
+            completion?()
+        }
+    }
+
+    /// Presents a Helpshift conversation window.
+    ///
+    /// - Parameters:
+    ///   - viewController: The view controller from which to present the Helpshift window.
+    ///   - refreshUserDetails: If `true`, refresh user ID, display name, and email emailAddress
+    ///     from the WordPress.com REST API (if appropriate) before displaying the window.
+    ///   - completion: Optional block to be called when the window is presented.
+    func presentHelpshiftConversationWindowFromViewController(viewController: UIViewController, refreshUserDetails: Bool, completion: (() -> Void)?) {
+        prepareToDisplayHelpshiftWindow(refreshUserDetails) {
+            HelpshiftSupport.showConversation(viewController, withOptions: self.optionsDictionary)
+            completion?()
+        }
+    }
+
+    /// Presents a Helpshift window displaying the FAQs index.
+    ///
+    /// - Parameters:
+    ///   - viewController: The view controller from which to present the Helpshift window.
+    ///   - refreshUserDetails: If `true`, refresh user ID, display name, and email emailAddress
+    ///     from the WordPress.com REST API (if appropriate) before displaying the window.
+    ///   - completion: Optional block to be called when the window is presented.
+    func presentHelpshiftFAQWindowFromViewController(viewController: UIViewController, refreshUserDetails: Bool, completion: (() -> Void)?) {
+        prepareToDisplayHelpshiftWindow(refreshUserDetails) {
+            HelpshiftSupport.showFAQs(viewController, withOptions: self.optionsDictionary)
+            completion?()
+        }
+    }
+
+    private func updateHelpshiftUserDetailsWithUserID(userID: String, displayName: String, emailAddress: String) {
+        HelpshiftSupport.setUserIdentifier(userID)
+        HelpshiftCore.setName(displayName, andEmail: emailAddress)
+    }
+
+    private func prepareToDisplayHelpshiftWindow(refreshUserDetails: Bool, completion: () -> Void) {
+        NSUserDefaults.standardUserDefaults().setBool(true, forKey: UserDefaultsHelpshiftWasUsed)
+
+        PushNotificationsManager.sharedInstance.registerForRemoteNotifications()
+        InteractiveNotificationsManager.sharedInstance.registerForUserNotifications()
+
+        let context = ContextManager.sharedInstance().newDerivedContext()
```

Comment:
I believe this is running on the main thread since it's initiated by UI. We should either use `mainContext` or do `context.performBlock` when using the context